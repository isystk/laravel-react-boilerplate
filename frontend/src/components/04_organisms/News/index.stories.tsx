import { Meta, Story } from '@storybook/react'
import React from 'react'
import News from './index'

export default {
  title: '04_organisms/News',
  component: News,
} as Meta

const Template: Story = () => {
  return <News />
}

export const Primary = Template.bind({})
Primary.storyName = 'プライマリ'
Primary.args = {}
