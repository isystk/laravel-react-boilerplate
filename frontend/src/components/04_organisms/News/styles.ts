import { css, keyframes } from '@emotion/css'

export const news = css``
