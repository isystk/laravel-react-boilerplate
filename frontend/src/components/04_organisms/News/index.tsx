import React, { FC } from 'react'
import { ContainerProps } from 'types'
import * as styles from './styles'
import { connect } from '@/components/hoc'
import ScrollIn from '@/components/02_interactions/ScrollIn'
import Image from '@/components/01_atoms/Image'

/** NewsProps Props */
export type NewsProps = Record<never, never>
/** Presenter Props */
export type PresenterProps = NewsProps

/** Presenter Component */
const NewsPresenter: FC<PresenterProps> = ({ ...props }) => (
  <>
    <section className={`${styles.news}`}>
      <div className="flex items-center bg-base justify-center w-full">
        <div className="py-32 w-full">
          <ScrollIn>
            <p className="text-black font-bold text-3xl md:mb-16 text-center">
              NEWS
            </p>
          </ScrollIn>
          <div className="md:px-64">
            <ScrollIn>
              <div className="flex flex-wrap items-center p-8 md:mb-16">
                <div className="flex w-full md:w-1/2 justify-end items-center">
                  <div className="w-full">
                    <p className="text-accent font-bold text-xl mb-4">
                      2022.04.22
                    </p>
                    <p className="font-bold break-words mb-4">
                      XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
                    </p>
                  </div>
                </div>
                <div className="flex w-full md:w-1/2 justify-center md:justify-start items-center">
                  <div className="mx-4 md:mx-0 md:ml-16 rounded-lg overflow-hidden">
                    <Image
                      src="/images/dummy_480x320.png"
                      alt="dummy"
                      lazy={false}
                      zoom={true}
                    />
                  </div>
                </div>
              </div>
            </ScrollIn>
            <ScrollIn>
              <div className="bg-gray-300 flex-none h-px"></div>
              <div className="flex flex-wrap items-center p-8 md:mb-16">
                <div className="flex w-full md:w-1/2 justify-end items-center">
                  <div className="w-full">
                    <p className="text-accent font-bold text-xl mb-4">
                      2022.04.22
                    </p>
                    <p className="font-bold break-words mb-4">
                      XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
                    </p>
                  </div>
                </div>
                <div className="flex w-full md:w-1/2 justify-center md:justify-start items-center">
                  <div className="mx-4 md:mx-0 md:ml-16 rounded-lg overflow-hidden">
                    <Image
                      src="/images/dummy_480x320.png"
                      alt="dummy"
                      lazy={false}
                      zoom={true}
                    />
                  </div>
                </div>
              </div>
            </ScrollIn>
            <div className="mb-36"></div>
          </div>
        </div>
      </div>
    </section>
  </>
)

/** Container Component */
const NewsContainer: React.FC<ContainerProps<NewsProps, PresenterProps>> = ({
  presenter,
  ...props
}) => {
  return presenter({
    ...props,
  })
}

export default connect<NewsProps, PresenterProps>(
  'News',
  NewsPresenter,
  NewsContainer
)
