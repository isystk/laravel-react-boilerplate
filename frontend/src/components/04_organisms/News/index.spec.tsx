import React from 'react'
import renderer from 'react-test-renderer'
import News from './index'

describe('News', () => {
  it('Match Snapshot', () => {
    const component = renderer.create(<News />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
