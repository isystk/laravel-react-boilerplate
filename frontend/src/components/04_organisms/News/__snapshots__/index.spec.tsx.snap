// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News Match Snapshot 1`] = `
<section
  className="css-0"
>
  <div
    className="flex items-center bg-base justify-center w-full"
  >
    <div
      className="py-32 w-full"
    >
      <div
        className="undefined css-pjt9eq css-1exhw58"
      >
        <p
          className="text-black font-bold text-3xl md:mb-16 text-center"
        >
          NEWS
        </p>
      </div>
      <div
        className="md:px-64"
      >
        <div
          className="undefined css-pjt9eq css-1exhw58"
        >
          <div
            className="flex flex-wrap items-center p-8 md:mb-16"
          >
            <div
              className="flex w-full md:w-1/2 justify-end items-center"
            >
              <div
                className="w-full"
              >
                <p
                  className="text-accent font-bold text-xl mb-4"
                >
                  2022.04.22
                </p>
                <p
                  className="font-bold break-words mb-4"
                >
                  XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
                </p>
              </div>
            </div>
            <div
              className="flex w-full md:w-1/2 justify-center md:justify-start items-center"
            >
              <div
                className="mx-4 md:mx-0 md:ml-16 rounded-lg overflow-hidden"
              >
                <div>
                  <img
                    alt="dummy"
                    className="css-s08p0c zoom"
                    src="/images/dummy_480x320.png"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="undefined css-pjt9eq css-1exhw58"
        >
          <div
            className="bg-gray-300 flex-none h-px"
          />
          <div
            className="flex flex-wrap items-center p-8 md:mb-16"
          >
            <div
              className="flex w-full md:w-1/2 justify-end items-center"
            >
              <div
                className="w-full"
              >
                <p
                  className="text-accent font-bold text-xl mb-4"
                >
                  2022.04.22
                </p>
                <p
                  className="font-bold break-words mb-4"
                >
                  XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
                </p>
              </div>
            </div>
            <div
              className="flex w-full md:w-1/2 justify-center md:justify-start items-center"
            >
              <div
                className="mx-4 md:mx-0 md:ml-16 rounded-lg overflow-hidden"
              >
                <div>
                  <img
                    alt="dummy"
                    className="css-s08p0c zoom"
                    src="/images/dummy_480x320.png"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="mb-36"
        />
      </div>
    </div>
  </div>
</section>
`;
